File: examples/sample.c
/* One-file C submission. Reads the instance from standard input. */
#include <stdio.h>
#include <string.h>

int main(void) {
    char line[256];
    int mode = 0, x = 0, y = 0, deadline = 0, max_x = 0;
    int have_patient = 0, hospitals = 0, first_available = 0;
    while (fgets(line, sizeof line, stdin)) {
        if (strncmp(line, "person", 6) == 0) { mode = 1; continue; }
        if (strncmp(line, "hospital", 8) == 0) { mode = 2; continue; }
        int px, py, pd;
        if (mode == 1 && sscanf(line, "%d,%d,%d", &px, &py, &pd) == 3) {
            if (!have_patient) { x = px; y = py; deadline = pd; max_x = px; have_patient = 1; }
            if (px > max_x) max_x = px;
        } else if (mode == 2) {
            int count;
            if (sscanf(line, "%d", &count) == 1) {
                hospitals++;
                if (count > 0 && first_available == 0) first_available = hospitals;
            }
        }
    }
    if (!have_patient) return 0;
    int hospital_x = max_x + 1;
    for (int i = 1; i <= hospitals; i++) {
        printf("H%d:%d,%d\n", i, hospital_x, y);
        fflush(stdout);
    }
    if (first_available && 2 * (hospital_x - x) + 2 <= deadline) {
        printf("0 H%d P1 H%d\n", first_available, first_available);
        fflush(stdout);
    }
    return 0;
}

File: examples/sample.jl
# One-file Julia submission. Reads the 2023 input from standard input.
mode = 0
x = 0
y = 0
deadline = 0
have_patient = false
hospitals = 0
first_available = 0

for raw in eachline(stdin)
    line = strip(raw)
    if startswith(line, "person")
        global mode = 1
    elseif startswith(line, "hospital")
        global mode = 2
    elseif mode == 1 && !have_patient
        fields = split(line, ',')
        if length(fields) == 3
            global x = parse(Int, fields[1])
            global y = parse(Int, fields[2])
            global deadline = parse(Int, fields[3])
            global have_patient = true
        end
    elseif mode == 2 && !isempty(line)
        count = parse(Int, line)
        global hospitals += 1
        if count > 0 && first_available == 0
            global first_available = hospitals
        end
    end
end

if have_patient
    for i in 1:hospitals
        println("H$i:$x,$y")
        flush(stdout)
    end
    if first_available > 0 && deadline >= 2
        println("0 H$first_available P1 H$first_available")
        flush(stdout)
    end
end
